build.rs

```diff
@@ -0,0 +1,116 @@
+#[macro_use]
+extern crate lazy_static;
+
+use std::fmt;
+use std::path::Path;
+use std::fs::{File, read_dir, metadata};
+use std::io::{Write, BufWriter};
+
+// Location of the dist folder
+#[cfg(not(target_os = "windows"))]
+const DIST: &str = "frontend/dist/";
+#[cfg(target_os = "windows")]
+const DIST: &str = "frontend\\dist\\";
+
+// Where to write the Assets out to
+#[cfg(not(target_os = "windows"))]
+const ASSETOUT: &str = "src/asset.in";
+#[cfg(target_os = "windows")]
+const ASSETOUT: &str = "src\\asset.in";
+
+// Ignore directories specified here
+#[cfg(target_os = "windows")]
+lazy_static! {
+    static ref IGNORE: Vec<&'static str> = {
+        vec![
+            "frontend\\dist\\ember-fetch"
+        ]
+    };
+}
+#[cfg(not(target_os = "windows"))]
+lazy_static! {
+    static ref IGNORE: Vec<&'static str> = {
+        vec![
+            "frontend/dist/ember-fetch"
+        ]
+    };
+}
+
+/// Find the assets and write out the corresponding files
+pub fn main() {
+    write_asset_file(acquire_assets());
+}
+
+/// Start the recursion from the top level of the frontend's dist folder
+fn acquire_assets() -> Vec<Asset> {
+    recurse_dir(&Path::new(DIST))
+}
+
+/// Recurse through a directory, if it's an asset file add it to our list and if
+/// it's a directory get all the assets in there then put the `Assets` returned
+/// into this directory's Vector and return that.
+fn recurse_dir<P>(path: &P) -> Vec<Asset>
```

Comment:
yeah it also handles tricky things like "recurse so many times you blow the stack" and such 😄